src/coreclr/nativeaot/System.Private.CoreLib/src/System.Private.CoreLib.csproj

```
@@ -94,6 +94,7 @@
     <Compile Include="Internal\Runtime\CompilerHelpers\StartupCode\StartupCodeHelpers.Extensions.cs" />
     <Compile Include="Internal\Runtime\CompilerHelpers\ArrayHelpers.cs" />
     <Compile Include="Internal\Runtime\CompilerHelpers\InteropHelpers.cs" />
+    <Compile Include="Internal\Runtime\CompilerHelpers\InteropHelpers.macOS.cs" Condition="'$(FeatureObjCMarshal)' == 'true'" />
```

```suggestion
    <Compile Include="Internal\Runtime\CompilerHelpers\InteropHelpers.OSX.cs" Condition="'$(FeatureObjCMarshal)' == 'true'" />
```
The CoreLib project is using OSX extension for macOS specific files.

---

Resolved by removing the file entirely.